### PR TITLE
GUACAMOLE-857:  Add guacamole-auth-header to Docker image

### PR DESCRIPTION
**Pull request by jsf9k**
Thanks for providing such a useful piece of software!

This pull request adds `guacamole-auth-header` to the `guacamole-client` Docker image.  Please let me know what I need to do to shepherd this pull request through the approval process.

The addition of `guacamole-auth-header` is very useful for folks like me who are running a Dockerized Guacamole instance behind a proxy that authenticates and authorizes with FreeIPA.

### CODE REVIEW

**Comment by felddy** [comment]
Strong work 💪 @jsf9k 

This is PR solves the problem we were having. 🎉

**Comment by necouchman** [comment]
@jsf9k: Definitely appreciate the contribution, but please limit a pull request to a single JIRA issue.

**Comment by necouchman** [comment]
Also, I just responded on #514 - it looks pretty good, just needs a couple of tweaks to it.  So, I'd limit this PR to handling GUACAMOLE-857.

**Comment by jsf9k** [comment]
@necouchman please have another look.  I removed the CAS-related foo and force pushed.  I also updated the title to remove mention of GUACAMOLE-1082.

**Comment by necouchman** [comment]
@jsf9k: Okay, commit messages need to be tagged with JIRA issue, along with pull request...

**Comment by jsf9k** [comment]
> @jsf9k: Okay, commit messages need to be tagged with JIRA issue, along with pull request...

@necouchman I tagged the commit messages with the JIRA issue and force pushed.  I don't _think_ there is anything else listed in the CONTRIBUTING document that I have failed to do, but please let me know if there is.  Apologies for not reading that document the first time.

File: guacamole-docker/bin/start.sh [hunk 1]
```diff
@@ -591,6 +591,18 @@ END
     ln -s /opt/guacamole/duo/guacamole-auth-*.jar   "$GUACAMOLE_EXT"
 }
 
+##
+## Adds properties to guacamole.properties which configure the header
+## authentication provider.
+##
+associate_header() {
+    # Update config file
+    set_optional_property   "http-auth-header"                 "$HTTP_AUTH_HEADER"
```
[the remark this answers is not in the document]
Comment (jsf9k) [reply guacamole-docker/bin/start.sh]: Done.  See commit 8be628a.  I did another force-push since a fixup commit made the most sense for this.

**Comment by jsf9k** [comment]
@necouchman is there some CI foo that will create automatically create a new Docker image and push it up to Docker Hub?  How can I know when that happens, short of polling Docker Hub?

The reason I'm asking is that I want to switch from my `cisagov/guacamole-client` Docker image back to the official one.

Thanks again for guiding me through the approval process!

**Comment by necouchman** [comment]
@jsf9k Currently Docker images are only pushed when we release a version, so you'll have to wait for the next version.  We've had some discussions about doing nightly builds, but haven't done anything with that, yet.

**Comment by jsf9k** [comment]
@necouchman, is there any hope for a new release soon?  It's been six months.

(I'd actually forgotten all about this.  My colleague asked me why we were using our own `cisagov/guacamole-client` Docker image and I started retracing my steps.  :smile:)

**Comment by necouchman** [comment]
> Is there any hope for a new release soon? It's been six months.

We're finishing things up for 1.3.0 - looks like Mike identified a couple of regressions that need to be fixed, and then we have all of the administrative work (generating release notes, releasing the candidate, voting, etc.). Probably another couple of weeks.

**Comment by jsf9k** [comment]
I see that 1.3.0 is out now, and it contains these changes.  Huzzah!

**Pull request merged**